src/streaming/Stream.js

```diff
@@ -430,6 +434,29 @@ function Stream(config) {
         }
     }
 
+    function filterCodecs(type) {
+        let realAdaptation = dashManifestModel.getAdaptationForType(manifestModel.getValue(), streamInfo.index, type, streamInfo);
+        let codec,
+                i;
+
+        if (!realAdaptation || !Array.isArray(realAdaptation.Representation_asArray)) return null;
+
+        // Filter codecs that are not supported
+        // But keep at least codec from lowest representation
+        i = 1;
```

Comment:
What do you think if we do the same but using filter method of the array? I think that could help to better understand it. 

Ex:
```
realAdaptation.Representation_asArray.filter((_, i) => {
	if (i === 0) return true;

	codec = dashManifestModel.getCodec(realAdaptation, i);
	if (codec && !capabilities.supportsCodec(VideoModel(context).getInstance().getElement(), codec)) {
		log('[Stream] codec not supported: ' + codec);
		return false;
	}
	return true;
})
```

Reply:
Hi @epiclabsDASH ,

indeed, you can add a commit with this modification. The filter method is already used in other parts of the code : not a problem for me! ;-)

Nicolas

Reply:
Sure, no problem. Just my thought but happy to add the commit myself. 

By the way, thanks for the PR, it is really useful.

Reply:
To add this was tested and works well. @epiclabsDASH I will leave it to you to commit filter and merge.  I can test again if you need it, just ping

Reply:
Done!

Many thanks